index.js

```diff
@@ -53,14 +53,24 @@ function ensureAllCommentsPrinted(astComments) {
   });
 }
 
-function format(text, opts, addAlignmentSize) {
+function formatWithCursor(text, opts, addAlignmentSize) {
   addAlignmentSize = addAlignmentSize || 0;
 
   const ast = parser.parse(text, opts);
 
   const formattedRangeOnly = formatRange(text, opts, ast);
   if (formattedRangeOnly) {
-    return formattedRangeOnly;
+    return { formatted: formattedRangeOnly };
+  }
+
+  let cursorOffset;
+  if (opts.cursorOffset >= 0) {
```

Comment:
(for future reference, the discussion about this is [above](https://github.com/prettier/prettier/pull/1637#discussion_r119736286))